thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/data/repository/QuestionsSetRepository.java

```diff
@@ -0,0 +1,18 @@
+package pl.edu.agh.quizzesthesis.data.repository;
+
+import org.springframework.data.repository.CrudRepository;
+import pl.edu.agh.quizzesthesis.data.entity.QuestionsSet;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface QuestionsSetRepository extends CrudRepository<QuestionsSet, String> {
+
+    Optional<QuestionsSet> findByQuestionsSetKey(String questionsSetKey);
```

Comment:
`questionSetKey `is an ID. `CrudRepository `comes with predefined method `findById`, so you don't have to create a new one that does the same.

Reply:
`CrudRepository` is just a black magic to me. I will gladly use your suggestion ;).